src/core/block.ts

```diff
@@ -0,0 +1,191 @@
+import { EventBus } from './event-bus';
+import { Props } from './types';
+import {toKebab} from "../utils";
+
+type Meta = {
+  tagName: string;
+  props: Props;
+};
+
+export enum EventsEnum {
+	INIT = 'init',
+	FLOW_CDM = 'flow:component-did-mount',
+	FLOW_CDU = 'flow:component-did-update',
+	FLOW_RENDER = 'flow:render',
+}
+
+export class Block<T> {
```

Comment:
То что здесь класс Block определен как дженерик никак не используется, нужно определить тип для props как` T`

```suggestion
export class Block<T extends {
  events?: Events;
  children?: Children;
}> {
  props: T;
```